src/eventgrid/azext_eventgrid/_params.py

```diff
@@ -59,23 +59,44 @@ def load_arguments(self, _):
         c.argument('input_schema', arg_type=get_enum_type(['eventgridschema', 'customeventschema', 'cloudeventv01schema'], default='eventgridschema'), help='Schema in which incoming events will be published for this domain. If customeventschema is specified, either input_mapping_default_values or input_mapping_fields must be specified as well.')
 
     with self.argument_context('eventgrid event-subscription') as c:
+        c.argument('topic_name', deprecate_info=c.deprecate(expiration='2.1.0', hide=True), help='Name of Event Grid topic', options_list=['--topic-name'], completer=get_resource_name_completion_list('Microsoft.EventGrid/topics'))
         c.argument('event_subscription_name', arg_type=name_type, help='Name of the event subscription')
         c.argument('event_delivery_schema', arg_type=get_enum_type(['eventgridschema', 'custominputschema', 'cloudeventv01schema']), help='The schema in which events should be delivered for this event subscription. By default, events will be delivered in the same schema in which they are published (based on the corresponding topic\'s input schema).')
         c.argument('max_delivery_attempts', help="Maximum number of delivery attempts. Must be a number between 1 and 30.")
         c.argument('event_ttl', help="Event time to live (in minutes). Must be a number between 1 and 1440.")
         c.argument('deadletter_endpoint', help="The Azure resource ID of an Azure Storage blob container destination where EventGrid should deadletter undeliverable events for this event subscription.")
 
     with self.argument_context('eventgrid event-subscription create') as c:
+        c.argument('source_resource_id', help="Fully qualified identifier of the Azure resource to which the event subscription needs to be created.")
+        c.argument('resource_id', deprecate_info=c.deprecate(redirect="--source-resource-id", expiration='2.1.0', hide=True))
```

Comment:
It would be good to extract this common info to a CliCommandType so that if you have to change it later, it is changed everywhere.

Reply:
Thanks for the feedback, I have updated this now to remove this duplication.